plugins/quickicon/eos/src/Extension/Eos.php

```diff
@@ -0,0 +1,382 @@
+                'messageText'   => 'PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING',
+                'quickiconText' => 'PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING_SHORT',
+                'messageType'   => 'warning',
+                'image'         => 'warning-circle',
+                'messageLink'   => 'https://docs.joomla.org/Special:MyLanguage/Planning_for_Mini-Migration_-_Joomla_3.10.x_to_4.x',
+                'groupText'     => 'PLG_QUICKICON_EOS_GROUPNAME_WARNING',
+                'snoozable'     => true,
+            ];
+        }
+        // We are in security only mode now, 12 month to go from now on
+        if ($monthsUntilEOS < 12) {
+            return [
+                'id'            => 3,
+                'messageText'   => 'PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY',
+                'quickiconText' => 'PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY_SHORT',
+                'messageType'   => 'warning',
+                'image'         => 'warning-circle',
+                'messageLink'   => 'https://docs.joomla.org/Special:MyLanguage/Planning_for_Mini-Migration_-_Joomla_3.10.x_to_4.x',
+                'groupText'     => 'PLG_QUICKICON_EOS_GROUPNAME_WARNING',
+                'snoozable'     => true,
+            ];
+        }
+        // We still have 16 month to go, lets remind our users about the pre upgrade checker
+        if ($monthsUntilEOS < 16) {
+            return [
+                'id'            => 2,
+                'messageText'   => 'PLG_QUICKICON_EOS_MESSAGE_INFO_02',
+                'quickiconText' => 'PLG_QUICKICON_EOS_MESSAGE_INFO_02_SHORT',
+                'messageType'   => 'info',
+                'image'         => 'info-circle',
+                'messageLink'   => 'https://docs.joomla.org/Special:MyLanguage/Pre-Update_Check',
+                'groupText'     => 'PLG_QUICKICON_EOS_GROUPNAME_INFO',
+                'snoozable'     => true,
+            ];
+        }
+        // Lets start our messages 2 month after the initial release, still 22 month to go
+        if ($monthsUntilEOS < 22) {
+            return [
+                'id'            => 1,
+                'messageText'   => 'PLG_QUICKICON_EOS_MESSAGE_INFO_01',
+                'quickiconText' => 'PLG_QUICKICON_EOS_MESSAGE_INFO_01_SHORT',
+                'messageType'   => 'info',
+                'image'         => 'info-circle',
+                'messageLink'   => 'https://www.joomla.org/4/#features',
+                'groupText'     => 'PLG_QUICKICON_EOS_GROUPNAME_INFO',
+                'snoozable'     => true,
+            ];
+        }
+
+        return false;
+    }
+
+    /**
+     * Check if current user is allowed to send the data
+     *
+     * @return  bool
+     *
+     * @since __DEPLOY_VERSION__
+     *
+     * @throws Exception
+     */
+    private function isAllowedUser(): bool
+    {
+        return Factory::getApplication()->getIdentity()->authorise('core.login.admin');
```

Comment:
```suggestion
        return $this->getApplication()->getIdentity()->authorise('core.login.admin');
```
use `$this->getApplication()`